src/main/java/oc/Eintrag.java

```diff
@@ -436,10 +439,44 @@ public void addWarlordTraits(String mandatoryChoice, boolean subfactionsAllowed,
         add(warlordTraits);
 	}
 	
+	public void addWeapons(Class<? extends RuestkammerVater> cls, boolean mandatoryChoice) {
+		weapons = new RuestkammerStarter(ID, randAbstand, cnt, cls, "");
+        weapons.getKammer().setType(name);
+        weapons.getKammer().setKeywords(keywords);
+        weapons.initKammer();
+        weapons.setButtonText("Waffen");
+        add(weapons);
+        weapons.setAbwaehlbar(!mandatoryChoice);
+	}
+	
+	public void addWeapons(Class<? extends RuestkammerVater> cls, boolean mandatoryChoice, String defaultFK, String defaultNK) {
+		weapons = new RuestkammerStarter(ID, randAbstand, cnt, cls, "");
+        weapons.getKammer().setType(name);
+        weapons.getKammer().setKeywords(keywords);
+        weapons.getKammer().setDefaultFK(defaultFK);
+        weapons.getKammer().setDefaultNK(defaultNK);
+        weapons.initKammer();
+        weapons.setButtonText("Waffen");
+        add(weapons);
+        weapons.setAbwaehlbar(!mandatoryChoice);
+	}
+	
+	public RuestkammerVater getWeapons() {
+		return weapons.getKammer();
+	}
+	
 	public void correctRuestkammerPosition(RuestkammerStarter ruestkammer, RuestkammerStarter reference){
 		ruestkammer.getPanel().setLocation(
                 (int) ruestkammer.getPanel().getLocation().getX(),
                 (int) reference.getPanel().getLocation().getY() + reference.getPanel().getSize().height + 5
         );
 	}
+
+	public Set<String> getKeywords() {
+		return keywords;
+	}
+
+	public void setKeywords(Set<String> keywords) {
```

Comment:
Maybe add as constructor argument. During the lifecycle of an `Eintrag` instance this property shouldn't change right?

Reply:
Since we only create instances of subclasses of `Eintrag` we can have constructor args on `Eintrag`. As long as the subclass has a zero args constructor we can initialize it.

Even if some units change these keywords dynamically I would find it more descriptive if you would provide *"static"* keywords on creation and then just toggle the other keywords with some add/remove functionality.

The visibility of the setter (and add/remove) method should also be `protected`.

the keywords themselves should also be an `enum` so that we don't have spelling issues.